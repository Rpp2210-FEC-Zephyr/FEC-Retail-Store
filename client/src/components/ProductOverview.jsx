import React, {useState, useEffect } from 'react';
import { TbStarsFilled } from "react-icons/tb";
import { IoAddCircleOutline} from "react-icons/io5";
import { GiHamburgerMenu} from "react-icons/gi";
import { BsCheck2Square} from "react-icons/bs";
import { IconContext } from "react-icons";
import { AiOutlineCheckCircle} from "react-icons/ai";
import { AiFillCheckCircle} from "react-icons/ai";
import RatingSystem from './RatingSystem.jsx'
import ReactDOM from 'react-dom'
import $ from 'jquery';

const Popup = require('../Notification.jsx')
const Favorites = require('./Favorites.jsx')



const ProductOverview = ({main}) =>{
  const [show, setShow] = useState([])
  const [style, setStyle] = useState([])
  const [skus, setSkus] = useState([])
  const [feature, setFeature] = useState([])
  const [cSize , setCSize] = useState(null)
  const [cQuant, setCQuant] = useState(null)
  const [cPrice, setCPrice] = useState(null)
  const [quantity, setQuantity] = useState([])

  const getStyles = (id) =>{

    $.ajax({
      type: 'GET',
      url: '/Styles',
      data: {id: id },
      success: (data) =>{
        console.log('Style Data', data)

        setStyle(data)
        setShow(data.results[0])
        setCPrice(data.results[0].original_price)
        setSkus(data.results[0].skus)
      }
    })

  }

  const getFeatures = (id) =>{

    $.ajax({
      type: 'GET',
      url: '/Features',
      data: {id: id },
      success: (data) =>{
        console.log('Feature Data', data.features)

        setFeature(data.features)

      }
    })
  }

  const onQuan = (quan) =>{

    quan = quan.split(' ')[0]

    quan = parseInt(quan)
    quan = quan + 1

    var results = []

    for(var i = 1; i < quan ; i++){
      results.push(i)
    }

    setQuantity(results)
    return


  }

  const onCSize = (csize) =>{
    csize = csize.split(' ')[1]

    if(csize == "Size"){
      setCSize(null)
      setCQuant(null)
    }else{
      setCSize(csize)
      setCQuant(1)
    }

  }
  const onCQuan = (cquan) =>{
    cquan = parseInt(cquan)
    setCQuant(cquan)


  }

  const Favor = () => {

     const Star = JSON.parse(localStorage.getItem('favorites'));
     if(Star != null){
      localStorage.setItem('favorites', JSON.stringify([Star, main]))
      Favorites.Toggle(main)

     }else{
      localStorage.setItem('favorites', JSON.stringify([ main]));
      Favorites.Toggle(main)
     }
  }


  const Change = (item, index) =>{
   const mySelect = document.getElementById(`${index}`);



    const Selecet = document.querySelectorAll('.StyleChoose')
    const check = document.querySelectorAll('.selected')


    Selecet.forEach(element =>{
    element.addEventListener('click', ()=> {


        check.forEach(el => {
        el.style.visibility = "hidden"
        })
  })
    })

  mySelect.style.visibility = "visible"


   setShow(item)
   Favorites.Status(main)
   setSkus(item.skus)


}

  const ImageExpander = () =>{

      setTimeout(() => {
        // Your logic here
        Popup.ImageExpander()
        Popup.Selected()

        ImageExpander();
      }, 1000);

  }




  const addBag = () =>{
    if(cSize == null || cQuant == null){

      Popup.Alert("error")

    }else{
      console.log('Current', show,cSize, cQuant)
      const bag = JSON.parse(localStorage.getItem('bag'));


      if(bag != null){
        if(bag.length == 10){
          Popup.Alert("warning")
        }else{
          Popup.Alert("success")
          localStorage.setItem('bag', JSON.stringify([bag, { cloth: show, size: cSize, quant: cQuant, name: main.name }]));
        }
      }else{
        Popup.Alert("success")
        localStorage.setItem('bag', JSON.stringify([{ cloth: show, size: cSize, quant: cQuant, name: main.name  }]));
      }

    }





  }







  useEffect(() =>{
    if(main.id != undefined){
      getStyles(main.id)
      getFeatures(main.id)

      ImageExpander()
      Popup.Selected()
      Favorites.Status(main)


    }




  },[main])



    return (
      <div class = 'productOverview'>

        <div class = 'left'>


    <div class="slider">
      <div class="slides">

        <input type="radio" name="radio-btn" id="radio1" />
        <input type="radio" name="radio-btn" id="radio2" />
        <input type="radio" name="radio-btn" id="radio3" />
        <input type="radio" name="radio-btn" id="radio4" />
        <input type="radio" name="radio-btn" id="radio5" />
        <input type="radio" name="radio-btn" id="radio6" />
        <input type="radio" name="radio-btn" id="radio7" />

        <div class="slide first">
          <img src= {show.length != 0 ? show.photos[0].url : null} alt="" />
        </div>
        {/* show.photos.slice(1, 4) */}
        {show.length != 0 ? show.photos.slice(1).map( (photo) => <div class = 'slide'>
        <img src= {photo.url} alt="" />
        </div>  ) : null}


        <div class = 'Popup'>
          <span>&times;</span>
          <img src = "" alt="" />

        </div>

        <div class="navigation-auto">
          <div class="auto-btn1"></div>
          <div class="auto-btn2"></div>
          <div class="auto-btn3"></div>
          <div class="auto-btn4"></div>
          <div class="auto-btn5"></div>
          <div class="auto-btn6"></div>
        </div>
      </div>

      <div class="navigation-manual">
        <label for="radio1" class="manual-btn"></label>
        <label for="radio2" class="manual-btn"></label>
        <label for="radio3" class="manual-btn"></label>
        <label for="radio4" class="manual-btn"></label>
        <label for="radio5" class="manual-btn"></label>
        <label for="radio6" class="manual-btn"></label>
      </div>

    </div>
    <div class ='slogan'>{main.slogan}</div>
    <div class = 'desc'>{main.description}</div>


        </div>

        <div class = 'right'>
          <div class = 'productReview'>
         <RatingSystem obj = {{rating: 4.4}}/>
         <div class = 'scroll'>
            Show All Reviews
         </div>
         </div>
          <div class = 'category'>
          {main ? main.category : null}
          </div>
          <div class = 'ProductName'>
            {main ? main.name : null}
          </div>
          <div class = 'price'>
            {main ? `$${cPrice}` : null}
          </div>

          <div class = 'style'>
            STYLE >   <div class = 'styleSelect'> SELECTED STYLE </div>
          </div>

          <div class = 'StyleContainer'>

            {style.length != 0 ? style.results.map((item, index) =>

              <div onClick = {(e) => {Change(item, index)}} class = 'StyleChoose'>
              <div class = 'selected' id = {index} style = {index == 0 ? {visibility: "visible"} : {visibility: "hidden"} }>
            <IconContext.Provider value={{ color: "#40D3DC", size:"20px" }}>
              <AiFillCheckCircle />
              </IconContext.Provider>
              </div>
                <div class = 'textBox'>
                 <p class = 'text head'>{item.name}</p>
                 </div>
                 <img src= {item.photos[0].url} alt="" />
                  </div>
              ):null}

          </div>
          <div class = 'select'>

          <select onChange = {(e) =>{onQuan(e.target.value); onCSize(e.target.value)}}   class="SelectSize" id="Sizes">
            <option selected="selected" >
              Choose Size
            </option>
            {skus.length !=0 ? Object.keys(skus).map((size) => <option value = {`${skus[size].quantity} ${skus[size].size}`}>{skus[size].size} </option>): null}
          </select>
          <select  onChange = {(e) =>{onCQuan(e.target.value)}}class="SelectNumber" id="Number">
            {quantity.length !=0 ? quantity.map((num) =><option value = {num}>{num} </option>) : null}

          </select>

          </div>
          <div class = 'Bag'>
          <ul class="notifications"></ul>
          <div class="buttons">

            <button onClick={() =>{addBag()}} class = 'Add buttons btn' > <div class = 'addComp'>
              Add To Bag
              </div>
              <div class = 'icon'>
              <IconContext.Provider value={{ color: "white", size:"40px" }}>
              < IoAddCircleOutline  />
              </IconContext.Provider>
              </div>
              </button>
              </div>

          <input type="checkbox" id="star" />
<label onClick = {() =>{
  Favor()
}}class = 'label' id="info" for="star">

  <svg viewBox="0 0 24 24">
    <path d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"></path>
  </svg>
</label>


          </div>
          <div>

          </div>

         <div class = 'feature'>
         <div class="vertical-divider"> ㅤ</div>

          <ul class  = 'featurelist'>
            {feature.length !=0 ? feature.map((feat) => <li class = 'featureitem'>
              <BsCheck2Square /> ㅤ
              {feat.value}
              </li>) : null}
          </ul>
         </div>


        </div>

        </div>

    )
  }

export default ProductOverview;
