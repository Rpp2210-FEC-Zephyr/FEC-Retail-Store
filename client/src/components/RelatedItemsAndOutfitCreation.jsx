import React, {useState, useEffect } from 'react';
import ReactDOM from 'react-dom'
import $ from 'jquery';



const RelatedItemsAndOutfitCreation = () =>{


    return (
      <div>RelatedItemsAndOutfitCreation</div>

    )
  }

export default RelatedItemsAndOutfitCreation;