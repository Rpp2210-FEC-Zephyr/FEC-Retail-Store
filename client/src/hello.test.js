import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import $ from 'jquery';
import ProductOverview from './components/ProductOverview';
import RelatedItemsAndOutfitCreation from './components/RelatedItemsAndOutfitCreation.jsx';

describe('ProductOverview component', () => {
  test('renders main product name', () => {
    const mainProduct = { id: 71697, name: 'Camo Onesie' };
    const Outfits = jest.fn();
    render(<ProductOverview main={mainProduct} Outfits={Outfits} />);
    expect(screen.getByText(/camo onesie/i)).toBeInTheDocument();
  });

  test('selects a size and quantity', async () => {
      const mainProduct = { id: 71697, name: 'Camo Onesie' };
    const Outfits = jest.fn();
      const tree = renderer.create(<ProductOverview main={mainProduct} Outfits={Outfits} />).toJSON();
      expect(tree).toMatchSnapshot();


});
})

jest.mock('jquery', () => ({
  ajax: jest.fn()
}));

describe('RelatedItemsAndOutfitCreation', () => {
  test('displays related items', async () => {
    const mockRelatedItems = [
      { id: 1, name: 'Related Item 1' },
      { id: 2, name: 'Related Item 2' },
      { id: 3, name: 'Related Item 3' },
    ];

    const mockMain = { id: 123 };

    const mockURL = 'http://example.com';

    // Mock the AJAX response from the server
    $.ajax.mockImplementationOnce((options) => {
      if (options.url === '/Related') {
        options.success(mockRelatedItems);
      }
    });

    // Render the component
    render(<RelatedItemsAndOutfitCreation main={mockMain} URL={mockURL} />);

    // Wait for the related items to be displayed
    await waitFor(() => {
      const items = screen.queryAllByTestId('related-item');
      expect(items).toHaveLength(0);
    });

    // Verify that the related items are displayed correctly
    mockRelatedItems.forEach((item) => {
      expect(screen.getByText(item.name)).toBeInTheDocument();
    });

    // Verify that the AJAX request was made with the correct parameters
    expect($.ajax).toHaveBeenCalledWith({
      type: 'GET',
      url: '/Related',
      data: { id: mockMain.id },
      success: expect.any(Function)
    });
  });
});

