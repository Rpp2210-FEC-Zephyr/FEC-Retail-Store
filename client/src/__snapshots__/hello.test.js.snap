// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductOverview component selects a size and quantity 1`] = `
<div
  className="productOverview"
>
  <div
    className="left"
  >
    <div
      className="slider"
    >
      <div
        className="slides"
      >
        <div
          className="slide first"
        >
          <img
            alt=""
            src={null}
          />
        </div>
        <div
          className="Popup"
        >
          <span>
            ×
          </span>
          <img
            alt=""
            src=""
          />
        </div>
        <div
          className="navigation-auto"
        />
      </div>
      <div
        className="navigation-manual"
      />
    </div>
    <div
      className="slogan"
    />
    <div
      className="desc"
    />
  </div>
  <div
    className="right"
  >
    <div
      className="productReview"
    >
      <div
        className="scroll"
      >
        Show All Reviews
      </div>
    </div>
    <div
      className="category"
    />
    <div
      className="ProductName"
    >
      Camo Onesie
    </div>
    <div
      className="price"
    >
      <div>
        $
         
      </div>
    </div>
    <div
      className="style"
    >
      STYLE &gt;   
      <div
        className="styleSelect"
      >
         
         
      </div>
    </div>
    <div
      className="StyleContainer"
    />
    <div
      className="select"
    >
      <select
        className="SelectSize"
        id="Sizes"
        onChange={[Function]}
      >
        <option
          defaultValue="selected"
        >
          Choose Size
        </option>
      </select>
      <select
        className="SelectNumber"
        id="Number"
        onChange={[Function]}
      />
    </div>
    <div
      className="Bag"
    >
      <ul
        className="notifications"
      />
      <div
        className="buttons"
      >
        <button
          className="Add buttons btn"
          onClick={[Function]}
        >
           
          <div
            className="addComp"
          >
            Add To Bag
          </div>
          <div
            className="icon"
          >
            <svg
              fill="currentColor"
              height="40px"
              stroke="currentColor"
              strokeWidth="0"
              style={
                {
                  "color": "white",
                }
              }
              viewBox="0 0 512 512"
              width="40px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M448 256c0-106-86-192-192-192S64 150 64 256s86 192 192 192 192-86 192-192z"
                fill="none"
                strokeMiterlimit="10"
                strokeWidth="32"
              />
              <path
                d="M256 176v160m80-80H176"
                fill="none"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="32"
              />
            </svg>
          </div>
        </button>
      </div>
      <input
        id="star"
        type="checkbox"
      />
      <label
        className="label"
        htmlFor="star"
        id="info"
        onClick={[Function]}
      >
        <svg
          viewBox="0 0 24 24"
        >
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
        </svg>
      </label>
    </div>
    <div />
    <div
      className="feature"
    >
      <div
        className="vertical-divider"
      >
         ㅤ
      </div>
      <ul
        className="featurelist"
      />
    </div>
  </div>
</div>
`;
